src/common/utils.ts

```diff
@@ -1,11 +1,39 @@
-export function permutations<T>(
-  xs: ReadonlyArray<T>
+/**
+ *
+ * <!--
+ * !@consider?
+ * - In terms of BoxedExpressions - optimizations which are always desirable to take place are
+ * possible...
+ *  ^Perhaps then, a wrapper BoxedExpr. utility for specifying these permutations via 'condition'
+ *  would be apt...?
+ *
+ * - ^If wishing to take adv. of this, the 'condition' callback would likely benefit from a second parameter typed as a collection
+ * ('Set' if enforcing unique) with all hitherto (arbitrary representations) of generated
+ * permutations.
+ *  (See commented snippets within function signature below.)
+ * -->
+ *
+ * @export
+ * @template T
+ * @param xs
+ * @param [condition]
+ * @returns
+ */
+export function permutations<T /* , Y extends any = any */>(
+  xs: ReadonlyArray<T>,
+  condition?: (
```

Comment:
Thought this made sense (see call from `matchPermutation` beneath), but feel free to revert if feeling it doesn't belong